kin-sdk/kin-sdk-lib/src/main/java/kin/sdk/Transaction.java

```diff
@@ -1,65 +1,61 @@
 package kin.sdk;
 
-import java.math.BigDecimal;
+import java.io.IOException;
 import kin.base.KeyPair;
+import kin.base.Memo;
+import kin.base.Network;
 
 public class Transaction {
 
-    private final KeyPair destination;
-    private final KeyPair source;
-    private final BigDecimal amount;
-    private final int fee;
-    private final String memo;
-
     /**
      * The transaction hash
      */
     private final TransactionId id;
 
-    private final kin.base.Transaction stellarTransaction;
-    private final WhitelistableTransaction whitelistableTransaction;
+    private final kin.base.Transaction baseTransaction;
 
-    Transaction(KeyPair destination, KeyPair source, BigDecimal amount, int fee, String memo,
-                TransactionId id, kin.base.Transaction stellarTransaction, WhitelistableTransaction whitelistableTransaction) {
-        this.destination = destination;
-        this.source = source;
-        this.amount = amount;
-        this.fee = fee;
-        this.memo = memo;
+    public Transaction(TransactionId id, kin.base.Transaction baseTransaction) {
         this.id = id;
-        this.stellarTransaction = stellarTransaction;
-        this.whitelistableTransaction = whitelistableTransaction;
+        this.baseTransaction = baseTransaction;
     }
 
-    public KeyPair getDestination() {
-        return destination;
+    public static Transaction decodeTransaction(String transactionEnvelope) throws IOException {
+        kin.base.Transaction transaction = kin.base.Transaction.fromEnvelopeXdr(transactionEnvelope);
+        TransactionId id = new TransactionIdImpl(Utils.byteArrayToHex(transaction.hash()));
+        return new Transaction(id, transaction);
     }
 
     public KeyPair getSource() {
-        return source;
-    }
-
-    public BigDecimal getAmount() {
```

Comment:
why did you remove getAmount() ?

Reply:
cause now it's a general transaction and not just a payment like before, and there might be no amount.
- need to think how to not break existing API
- consider having subclass for `PaymentTransaction` and `RawTransaction` maybe (like we have in server sdks)
- missing some fields here: hash, signature, operations (compare to base transaction object)

Reply:
Please see my comment above.
As for the other fields, i thought that we don't wont to exposed them yet...